### PR TITLE
switch CUDA code to main cmpflx routine

### PR DESCRIPTION
**Pull request by zingale**
This uses the main `cmpflx` machinery for the CUDA MOL. 

At the moment it crashes, likely because of the change in the dimensioning of `qint`.  We need to add a call to `store_godunov_state()`

### CODE REVIEW

**Comment by zingale** [comment]
one of the tests is failing currently, so this is not ready yet

**Comment by zingale** [comment]
tests are running

**Comment by zingale** [comment]
GPU diffs are roundoff:

http://groot.astro.sunysb.edu/Castro/test-suite/gpu/2019-01-15-003/index.html

**Comment by zingale** [comment]
these are the test result for cmpflx:
http://groot.astro.sunysb.edu/Castro/test-suite/gfortran/2019-01-17-006/index.html

File: Source/hydro/Castro_mol_nd.F90 [hunk 1]
```diff
@@ -596,22 +598,32 @@ subroutine ca_construct_flux_cuda(lo, hi, domlo, domhi, dx, dt, idir, &
...
     real(rt), intent(inout) :: flux(f_lo(1):f_hi(1), f_lo(2):f_hi(2), f_lo(3):f_hi(3), NVAR)
     real(rt), intent(in   ) :: area(a_lo(1):a_hi(1), a_lo(2):a_hi(2), a_lo(3):a_hi(3))
     real(rt), intent(in   ) :: dx(3)
     real(rt), intent(in   ), value :: dt
 
     !$gpu
 
-    call cmpflx_cuda(lo, hi, domlo, domhi, idir, qm, qm_lo, qm_hi, qp, qp_lo, qp_hi, &
-         qint, qe_lo, qe_hi, flux, f_lo, f_hi, qaux, qa_lo, qa_hi)
+    !call cmpflx_cuda(lo, hi, domlo, domhi, idir, qm, qm_lo, qm_hi, qp, qp_lo, qp_hi, &
```
Comment (maxpkatz) [inline Source/hydro/Castro_mol_nd.F90]: Can you remove this commented out code please?

**Pull request merged**